src/components/ciConfig/CIConfigForm.tsx

```diff
@@ -73,8 +73,28 @@ export default function CIConfigForm({
                     (material) => material.id === ciConfig?.ciBuildConfig?.buildContextGitMaterialId,
                 )
               : sourceConfig.material[0]
-    const currentBuildContextGitMaterial = buildCtxGitMaterial || currentMaterial
-    const [selectedMaterial, setSelectedMaterial] = useState(currentMaterial)
+
+    const getParsedCurrentMaterial = (material?): SelectedGitMaterialType => {
+        const _currentMaterial: SelectedGitMaterialType = {
+            ...(material ? material : currentMaterial),
+            value: currentMaterial.checkoutPath,
+            label: currentMaterial.name,
+            startIcon: getGitProviderIcon(currentMaterial.url),
+        }
+        return _currentMaterial
+    }
+
+    const getParsedSourceConfig = (): SourceConfigType => {
+        const _sourceConfig = { ...sourceConfig }
+        _sourceConfig.material = _sourceConfig.material.map((_material) => {
```

Comment:
Can we directly write it as 
```suggestion
        _sourceConfig.material = _sourceConfig.material.map(getParsedCurrentMaterial(_material))
```